test/AutoRemediateS3-002.test.js

```diff
@@ -0,0 +1,230 @@
+"use strict";
+
+const source = require('../functions/AutoRemediateS3-002');
+
+//const config = require('./config');
+const AWS = require("aws-sdk");
+
+const CCRuleCode = 'S3-002';
+const CCRuleName = 'BucketPublicReadAcpAccess';
+const allUsersURI = 'http://acs.amazonaws.com/groups/global/AllUsers';
+const readAcpPermission = "READ_ACP";
+const aclSkeleton = '{"Owner":"", "Grants":[]}'; // skeleton for new permission grants
+
+describe('S3-002 READ_ACP auto-remediation', () => {
+
+    describe('S3 ACL Grants shall transfer to new ACL except for READ_ACP grant', () => {
+
+        describe('S3 ACL with READ_ACP grant', () => {
+            it('is not transfered to new ACL', () => {
+                var aclNew = JSON.parse(aclSkeleton);
+                const readAcpGrant = JSON.parse('{ "Grantee": { "Type": "Group", "URI": "http://acs.amazonaws.com/groups/global/SomeUsers" }, "Permission": "READ_ACP" }');
+
+                expect(readAcpGrant.Permission).toBe('READ_ACP');
+                expect(source.removeAcpPermission(readAcpGrant, aclNew)).toEqual(JSON.parse(aclSkeleton));
+            });
+        });
+
+        describe('S3 ACL with READ grant', () => {
+            it('is transfered to new ACL', () => {
+                var aclNew = JSON.parse(aclSkeleton);
+                const readGrant = JSON.parse('{ "Grantee": { "Type": "Group", "URI": "http://acs.amazonaws.com/groups/global/AllUsers" }, "Permission": "READ" }');
+                const readAcl = JSON.parse('{ "Owner":"", "Grants":[ { "Grantee": { "Type": "Group", "URI": "http://acs.amazonaws.com/groups/global/AllUsers" }, "Permission": "READ" } ] }');
```

Comment:
rename `readAcl` -> `expectedAcl`.

Makes the intention clearer and easier to read.

Can to re-use `readGrant` to make it more readable and DRY?

```
const readAcl = { Owner:"", Grants:[ readGrant ] };
```